### ISSUE TITLE
Prevent default sub-document on object creation

### ISSUE THREAD

**Issue by jeremyml** [Automattic/mongoose#6269]
Mongoose version 4.13.8

I was suprised to see a subdocument created when I create the parent with `findOneAndUpdate()` and options `setDefaultsOnInsert: true`.  My schema has no default value set for the field with the subdocument.

``` javascript
SubDocSchema = new mongoose.Schema({},{
  autoIndex : true, 
  toObject: { virtuals: true },
  toJSON: { virtuals: true },
});

ParentSchema = new mongoose.Schema({
  docVal: String,
  subDoc: SubDocSchema, // no default is set here!
},{
  autoIndex : true, 
  toObject: { virtuals: true },
  toJSON: { virtuals: true },
});

parentModel.findOneAndUpdate(
  {_id: parentId}, 
  {docVal: 'newDocVal'}, 
  {upsert: true, setDefaultsOnInsert: true}
).exec();
```

``` javascript
{
  _id: ObjectId("5ab55f0facdae000111d39c0"),
  docVal: 'newDocVal',
    subdoc: {
      _id: ObjectId("5ab55d59bf634e001091024c")
  }
}
```

Is it possible to prevent the subdocument without using `$unset`?

**Comment by varunjayaraman** [comment]
youre right, this seems to be a bug with `setDefaultsOnInsert`:

```javascript
const mongoose = require('mongoose');
mongoose.Promise = global.Promise;

run().catch(error => console.error(error.stack));

async function run() {
  await mongoose.connect(`mongodb://localhost:27017/gh-6269`);
  
  const subDocSchema = new mongoose.Schema({ name: String });

  const schema = new mongoose.Schema({
    subDoc: subDocSchema,
    test: String
  });

  const Model = mongoose.model('Model', schema);

  const fakeDoc = new Model({});
  const doc = await Model.create({});

  // toggle to false to see correct behavior
  // where subdoc is not created
  const setDefaultsFlag = true;

  console.log(await Model.findOneAndUpdate({ _id: fakeDoc._id }, {
    test: 'lol'
  }, { setDefaultsOnInsert: setDefaultsFlag, upsert: true, new: true }));

}
```

**Comment by vkarpov15** [comment]
Fix will be in 5.0.13, thanks for reporting :+1:

**Comment by jeremyml** [comment]
@vkarpov15 thank you for the work you do on Mongoose
